cider/exec/plan/validator/CiderPlanValidator.cpp

```diff
@@ -31,25 +31,33 @@
 namespace validator {
 
 bool CiderPlanValidator::validate(const substrait::Plan& plan,
-                                  const generator::FrontendEngine& from_platform) {
+                                  const generator::FrontendEngine& frontend_engine) {
   // Assumed that plan pattern already validated, do function look up first
   // Apply specific rule check on the plan
-  return isSupportedSlice(constructPlanSlice(plan, from_platform), from_platform);
+  return isSupportedSlice(
+      constructPlanSlice(plan, frontend_engine), plan, frontend_engine);
 }
 
 PlanSlice CiderPlanValidator::constructPlanSlice(
     const substrait::Plan& plan,
-    const generator::FrontendEngine& from_platform) {
+    const generator::FrontendEngine& frontend_engine) {
   const substrait::Rel& root_rel = getRootRel(plan);
   std::vector<substrait::Rel> rel_vec;
   putRelNodesInVec(root_rel, rel_vec);
-  return PlanSlice{rel_vec, from_platform};
+  return PlanSlice{rel_vec, frontend_engine};
 }
 
 bool CiderPlanValidator::isSupportedSlice(
     const PlanSlice& plan_slice,
-    const generator::FrontendEngine& from_platform) {
-  // TODO: (yma11) add function look up and rule based check
+    const substrait::Plan& plan,
+    const generator::FrontendEngine& frontend_engine) {
+  try {
```

Comment:
Seems some of plan_slice will forever throw CiderCompileException in SubstraitToRelAlgExecutionUnit, because plan_slice's field index is wrong and plan is not coherent ？